README.md

```diff
@@ -0,0 +1,134 @@
+one GiB is zeroed out to destroy existing partitions. This stage creates
+three partitions:
+1. GRUB (4 MiB)
+2. Swap (1020 MiB)
+3. The filesystem (rest)
+
+On GPT based partition tables GRUB needs its own partition for booting. 
+
+After partitioning the Swap partition is initialized and the root filesystem
+is formatted using the ext4 filesystem type. Auto checking is disabled both
+based on time and mount count.
+
+## Starting debootstrap
+Now its time to mount the root filesystem and install a minimal
+Debian/Ubuntu Linux based on _debootstrap_. This is the point where the
+scripts _create-debian.sh_ and _create-ubuntu.sh_ differs since the
+installation is minimal different for both distributions, e.g. the download
+URL.
+
+## Copying raw filesystem structure
+After installing the base system the contents of the directory
+_/root/deboot/_ is copied recursively using _rsync_. This directory also
+contains the post installation scripts which are called later using
+_chroot_. Feel free to add contents into that directory to customize to your
+needs such additional configuration files. It is a good place to add for
+example a proxy server configuration to the _apt_ subsystem.
+
+There is a post copy feature which replaces all _%DISTR%_ token occurences
+in the files with the real _DISTR_ environment variable given by parameter
+or by default from the file _/etc/default/kvm-hostcreator_. So it is
+possible to add additional _sources.list_ files which may contain the
+distribution.
+
+Finally the file _/etc/resolv.conf_ is copied into the VM to ensure same
+hostname resolution as of the host.
+
+## Postunpacking
+In this stage the file _/etc/hostcreator/postunpack.sh_ is called if
+existing. You can adjust this file as you need. It is a good place to add
+additional APT key downloads using the _gpg2_ tool. The precedure may differ
+depending on what software sites you use. The create scripts call this hook
+with the virtual hostname as the first parameter, the temporal mount point
+as the second parameter and the _DISTR_ variable as the third parameter.
+
+This is also the stage where the virtual hostname is configured and some
+useful aliases are added to the root users bash profile file.
+
+## Postinstallation
+There are some jobs needed in a chrooted environment on the newly created
+virtual machine. Now the create scripts start the
+_/root/deboot/root/bin/install.sh_ script which is already copied earlier
+through the raw copy stage and appears in the VM as file
+_/root/bin/install.sh_.
+
+The intention of that script is to configure date time and language
+settings. After that stage the APT subsystem has already updated the package
+revision database and upgraded pending updates.
+
+If the script _/root/bin/install-custom.sh_ exists in the filesystem it is
+called to add installation of packages you need additionally. So adjust the
+file _/root/deboot/root/bin/install-custom.sh_ prior VM creation on your
+demands.
+
+## Postconfiguration
```

Comment:
The postinstallation step lets me add additional packages via the _install-custom.sh_ script. This is done in the chroot environment.
Could you give an example what could be done in the postconfiguration step.   
Could this be the right place for setting up the root password or another user in the VM? Doing this requires chroot again, so why should we not do this in the install-custom.sh script?

Reply:
I have added some comments to the README.md that hopefully will help you.